source/server/listener_manager_impl.cc

```diff
@@ -266,9 +266,14 @@ Init::Manager& ListenerImpl::initManager() {
   }
 }
 
-void ListenerImpl::setSocket(const Network::ListenSocketSharedPtr& socket) {
+void ListenerImpl::setSocket(const Network::SocketSharedPtr& socket) {
   ASSERT(!socket_);
   socket_ = socket;
+  // Server config validation sets nullptr sockets.
+  if (socket_ && listen_socket_options_) {
+    bool ok = listen_socket_options_->setOptions(*socket_);
```

Comment:
Should this throw or fail in some way? This will also break NVLOG builds, so at minimum needs an `UNREFERENCED_PARAMATER(ok);`